sbkeys/README.md

```diff
@@ -0,0 +1,91 @@
+# Secure Boot Keys for Bottlerocket
+
+This document describes the tools available to generate the files needed for Secure Boot support in Bottlerocket.
+
+## Background
+
+For Secure Boot support, many different keys, certificates, and configuration files are required for building and publishing images.
+The [ArchWiki guide to Secure Boot](https://wiki.archlinux.org/title/Unified_Extensible_Firmware_Interface/Secure_Boot#Using_your_own_keys) covers the purpose of most of these files, along with sample commands to generate them.
+To keep build logic simple, a complete set of these files must be present for each build, and the files must follow the expected naming conventions.
+
+Each set of files is referred to as a Secure Boot Keys ("sbkeys") profile.
+The tools provided in this directory can be used to generate either a [local profile](#create-a-profile-with-local-resources) or an [AWS-based profile](#create-a-profile-with-aws-based-resources).
+If your preferred solution for key management is not supported, a contribution that adds a new tool or profile type would be welcome.
+
+To streamline the process of building Bottlerocket with Secure Boot support, a local profile will be generated automatically.
+This is done to minimize costs and to avoid requiring developers to set up infrastructure for key management ahead of time.
+However, because these local profiles offer direct access to private key materials, they are **strongly discouraged** for any kind of production use.
+
+Different profiles can be [specified at build time](#specify-a-profile-at-build-time) to use a custom set of keys.
+
+## Create a profile with local resources
+
+The `generate-local-sbkeys` tool can be used to create a local Secure Boot Keys profile.
+
+It uses `openssl` to generate private keys and certificate authorities (CAs), and `gpg` to create a GPG private key.
+It also uses [virt-fw-vars](https://pypi.org/project/virt-firmware/) to generate EFI variable data for the edk2 variable stores used by QEMU, and for registering Amazon EC2 AMIs.
```

Comment:
> for registering Amazon EC2 AMIs.

Is that accurate, the scripts call `virt-fw-vars` to output the EFI vars, but not to register AMIs (unless I misread the scripts).

Reply:
You understood correctly - I'll see if I can revise the wording. The EFI variable data is subsequently used for QEMU runs, and also used when registering AMIs.